### PR TITLE
[EBPF] added runtime compilation for GPU

### PR DESCRIPTION
**Pull request by val06**
### What does this PR do?

adds runtime compilation support for gpu-monitoring module

### Motivation

match the standard for all ebpf modules in system-probe

### Describe how to test/QA your changes

added UT for the runtime compilation
refactoring existing UTs to support build modes (separate [PR](https://github.com/DataDog/datadog-agent/pull/30846))
### Additional Notes

we deliberately decided not support prebuilt mode, as it is a deprecated mode

[Jira ticket](https://datadoghq.atlassian.net/browse/EBPF-591)

### CODE REVIEW

File: pkg/gpu/probe.go [hunk 1]
```diff
@@ -9,14 +9,11 @@ package gpu
 
 import (
 	"fmt"
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
```
Comment (gjulianm) [inline pkg/gpu/probe.go:12]: nit: I think these imports are not sorted properly.

File: pkg/gpu/probe.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ type ProbeDependencies struct {
 
 // Probe represents the GPU monitoring probe
 type Probe struct {
-	mgr            *ddebpf.Manager
+	m              *manager.Manager
```
Comment (gjulianm) [inline pkg/gpu/probe.go:49]: Why are we switching to use the unwrapped manager? We'll need it if we want to add telemetry to the eBPF probes, no?

File: pkg/gpu/ebpf/c/types.h [hunk 3]
```diff
@@ -1,4 +1,5 @@
-#pragma once
```
Comment (gjulianm) [inline pkg/gpu/ebpf/c/types.h:1]: :question: Is `#pragma once` not supported? Asking to know what to do for new files.

File: tasks/system_probe.py [hunk 4]
```diff
@@ -357,16 +357,16 @@ def ninja_test_ebpf_programs(nw: NinjaWriter, build_dir):
 
 
 def ninja_gpu_ebpf_programs(nw: NinjaWriter, co_re_build_dir: Path | str):
-    gpu_programs_co_re_dir = Path("pkg/gpu/ebpf/c")
-    gpu_programs_co_re_flags = f"-I{gpu_programs_co_re_dir}"
-    gpu_programs_co_re_programs = ["gpu"]
+    gpu_c_dir = Path("pkg/gpu/ebpf/c")
+    gpu_flags = f"-I{gpu_c_dir} -g"
+    gpu_programs = ["runtime/gpu"]
```
Comment (gjulianm) [inline tasks/system_probe.py:362]: I think this is making the build code place the object in the wrong folder, that's why the KMT tests are all red. Something like this 
```suggestion
    gpu_inc_dir = Path("pkg/gpu/ebpf/c")
    gpu_c_dir = gpu_inc_dir / "runtime"
    gpu_flags = f"-I{gpu_inc_dir} -g"
    gpu_programs = ["gpu"]
```

File: pkg/gpu/ebpf/c/types.h (hunk 3 above)
Comment (val06) [reply pkg/gpu/ebpf/c/types.h:1]: it is supported by the compiler we are using, but i just followed the same pattern as the rest of the .h files in our repo for parity

**Review by gjulianm — APPROVED** [review]

**Pull request merged**